feat(unity): Windows native support (#4825)
getsentry/sentry-docs: src/platforms/unity/native-support/index.mdx

sidebar_order: 100
---

> Currently, the Unity SDK offers native support for `Android`, `iOS`, and `Windows`. The support is `enabled` by default and you can opt out for each platform individually in the configuration window.

Reviewer: Same comment as above. For "enabled", since we're talking about a configuration setting specifically, code format is okay here, but the wording just has to be tweaked a bit to support that.

```suggestion
Currently, the Unity SDK offers native support for Android, iOS, and Windows. The support is set to `enabled` by default and you can opt out for each platform individually in the configuration window.
```

## Android & iOS

<Note>

> The setup for the iOS and Android native SDKs happens during build time with the options defined in the Sentry editor configuration window. Changes to the options object done through [ScriptableOptionsConfiguration](/platforms/unity/configuration/options/) will not affect events from the native layer.

Reviewer: ```suggestion
The setup for the iOS and Android native SDKs happens during build time, with the options defined in the Sentry editor configuration window. Changes to the options object done through [ScriptableOptionsConfiguration](/platforms/unity/configuration/options/) will not affect events from the native layer.
```

</Note>

The Unity SDK passes the options defined through the Sentry Unity Editor window during build time. For Android, it embeds the [Android SDK](/platforms/android/) inside the Gradle project created by Unity. The [iOS SDK](/platforms/apple/guides/ios/) is embedded inside the generated Xcode project. This means there's no need to use the Unity [built-in crash reporting](https://docs.unity3d.com/ScriptReference/CrashReport.html) functionality.

## Windows

> The Unity SDK initialized the Native SDK through the C# layer, and this means that we currently do not capture errors of the Unity engine itself.

Reviewer 2: I wonder if we can clarify a bit how it works (and its straights) before we point out this "Engine crashes" situation which most ppl don't come here expecting (it's kind of a unique feature we added on Android and iOS).

```suggestion
Windows native crashes are automatically captured through minidumps. Minidump capture is configured and enabled once the C# scripting layer runs, as early as possibly after Unity engine starts. This is slightly different than the Android and iOS support where we inject native initialization code that runs before Unity engine runs.
```
